native_client/definitions.mk

```diff
@@ -38,26 +38,32 @@ OS      := $(shell uname -s)
 
 # -Wl,--no-as-needed is required to force linker not to evict libs it thinks we
 # dont need ; will fail the build on OSX because that option does not exists
-ifneq ($(OS),Darwin)
-LDFLAGS += -Wl,--no-as-needed
+ifeq ($(OS),Linux)
+LDFLAGS_NEEDED := -Wl,--no-as-needed
+LDFLAGS_RPATH  := -Wl,-rpath,\$$ORIGIN -Wl,-rpath,.
+endif
+ifeq ($(OS),Darwin)
+LDFLAGS_NEEDED :=
+LDFLAGS_RPATH  := -Wl,-rpath,@executable_path -Wl,-rpath,.
```

Comment:
Are both `.` and `@executable_path` needed? I thought `.` would add the libraries with the paths from the build machine.